core/src/main/kotlin/com/exactpro/th2/read/db/core/impl/DataBaseServiceImpl.kt

```diff
@@ -83,16 +89,23 @@ class DataBaseServiceImpl(
                 emit(resultSet.transform(columns, associatedMessageType))
             }
         }.onCompletion { reason ->
-            reason?.also { LOGGER.warn(it) { "query $queryId completed with exception for $dataSourceId source" } }
+            try {
+                reason?.also { LOGGER.warn(it) { "query $queryId completed with exception for $dataSourceId source" } }
 
-            runCatching {
                 afterQueryHolders.forEach { holder ->
-                    execute(connection, holder, finalParameters)
+                    runCatching {
+                        execute(connection, holder, finalParameters)
+                    }.getOrElse {
+                        throw QueryExecutionException(
+                            "cannot execute ${holder.query} after query for $dataSourceId connection", it
```

Comment:
I am not sure if we should throw an exception when we have completed the main query execution. But if you want - I have no objections

Reply:
I think we should throw exception that show problem for users